docs/User/Tutorials/module-structure-and-quick-examples.md

```diff
@@ -123,18 +136,24 @@ but it adds failure handling and backtracking of input on failure.
 
 For example, to parse a sequence of digits:
 
-```haskell
+```haskell docspec
+>>> import Data.Function ((&))
 >>> import qualified Data.Char as Char
+>>> import qualified Streamly.Data.Stream.Prelude as Stream
+>>> import qualified Streamly.Data.Parser as Parser
+>>> import qualified Streamly.Data.Fold as Fold
+>>> import qualified Streamly.Internal.Data.Stream as Stream (parsePos)
 >>> decimal = Parser.takeWhile1 Char.isDigit Fold.toList
 >>> Stream.parsePos decimal $ Stream.fromList "1234 is the number"
 Right "1234"
 >>> Stream.parse decimal $ Stream.fromList "this is the number"
```

Comment:
use parsePos here to get correct position information instead of changing ParserError.